src/test/scala/mesosphere/marathon/integration/LeaderIntegrationTest.scala

```diff
@@ -50,4 +50,24 @@ class LeaderIntegrationTest extends IntegrationFunSuite
     And("the leader must have changed")
     WaitTestSupport.waitUntil("the leader changes", 30.seconds) { marathon.leader().value != leader }
   }
+
+  test("abdicate if the zk connection is dropped") {
+    Given("a leader")
+    WaitTestSupport.waitUntil("a leader has been elected", 30.seconds) { marathon.leader().code == 200 }
+
+    When("ZooKeeper dies")
+    ProcessKeeper.stopProcess("zookeeper")
+
+    Then("There is no leader in the cluster")
+    WaitTestSupport.waitUntil("a leader has been abdicated", 30.seconds) {
+      //TODO: should it be always 404?
+      Seq(404, 503).contains(marathon.leader().code)
+    }
+
+    When("Zookeeper starts again")
+    ProcessKeeper.startZooKeeper(config.zkPort, "/tmp/foo/single", wipeWorkDir = false)
```

Comment:
Maybe move it to a constant in SingleMarathonIntegrationTest to make that clear?